src/Extend/Theme.php

```diff
@@ -52,6 +53,54 @@ public function overrideFileSource(string $file, string $newFilePath, string $ex
         return $this;
     }
 
+    /**
+     * This method allows you to add custom Less functions.
+     *
+     * **Example usage:**
+     * ```php
+     * (new Extend\Theme)
+     *     ->addCustomLessFunction('is-flarum', function (mixed $text) {
+     *         return strtolower($text) === 'flarum'
+     *     }),
+     * ```
+     *
+     * @see https://leafo.net/lessphp/docs/#custom_functions
+     *
+     * @param string $functionName Name of the function identifier.
+     * @param callable $callable The PHP function to run when the Less function is called.
+     * @return self
+     */
+    public function addCustomLessFunction(string $functionName, callable $callable): self
+    {
+        $this->customFunctions[$functionName] = function (...$args) use ($callable) {
+            $argVals = array_map(function ($arg) {
+                return $arg->value;
+            }, $args);
+
+            $return = $callable(...$argVals);
+
+            // Booleans
+            if (is_bool($return)) {
+                return new \Less_Tree_Quoted('', $return ? 'true' : 'false');
+            }
+
+            // Numbers
+            if (! is_string($return) && is_numeric($return)) {
+                return new \Less_Tree_Dimension($return);
+            }
+
+            // Strings
+            if (is_string($return)) {
+                return new \Less_Tree_Quoted('', $return);
+            }
+
+            // Anything else
```

Comment:
I wonder if this should error instead of trying to cast: for example, arrays would cause problems here.